exonum/src/events/codec.rs

```diff
@@ -80,13 +79,30 @@ impl Decoder for MessagesCodec {
             );
         }
 
-        // Read message
-        if buf.len() >= total_len {
-            let data = buf.split_to(total_len).to_vec();
-            let raw = RawMessage::new(MessageBuffer::from_vec(data));
-            return Ok(Some(raw));
+        if total_len > buf.len() {
```

Comment:
This has been bothering me for a while: what if `total_len < buf.len()`, i.e., the message is undersized compared to its Noise wrapper? Do we accept such framing?

Reply:
After fixing `decrypted_msg_len` calculation we obtain the exact same message that we encrypted(without trailing zeros), so I think we can accept the only messages with total_len == buf.len().